src/hotspot/cpu/x86/x86.ad

```diff
@@ -1753,15 +1751,23 @@ const bool Matcher::match_rule_supported_vector(int opcode, int vlen, BasicType
     case Op_ClearArray:
     case Op_VectorMaskGen:
     case Op_VectorCmpMasked:
-    case Op_LoadVectorMasked:
-    case Op_StoreVectorMasked:
       if (!is_LP64 || !VM_Version::supports_avx512bw()) {
         return false;
       }
       if ((size_in_bits != 512) && !VM_Version::supports_avx512vl()) {
         return false;
       }
       break;
+    case Op_LoadVectorMasked:
+      if (!VM_Version::supports_avx512bw() && (is_subword_type(bt) || UseAVX < 1)) {
```

Comment:
With `UseAVX=0` we clear `supports_avx512bw`. So the test should be 
```
if (!VM_Version::supports_avx512bw() && is_subword_type(bt) || UseAVX < 1)
```

And may be naive question. Is VectorMaskGen is used for `mask` node  creation? If so, why to have separate support checks for `LoadVectorMasked/StoreVectorMasked`?

Reply:
Hi Vladimir, Existing expression gets benefit of short-circuiting else we will need to evaluate two expressions for truly supported case.

As of now VectorMaskGen is used for AVX3 targets, specially for partial in-lining for copy and vectorize compare and post vector loop processing. Partial inlining of copy is only enabled for sub-word types and for AVX2 we do not have sub-word handling yet,  for vectorized compare partial in-lining we use an explicit threshold i.e. array length is >= 16, thus only sub-word types and 512 bit integer species qualify this threshold.

I will be posting a subsequent patch with sub-word handling for masked load/stores over AVX2 after some performance analysis along with maskgen patterns for AVX2.